internal/actions/object/http_request.go

```diff
@@ -0,0 +1,70 @@
+package object
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/zitadel/zitadel/internal/actions"
+)
+
+// HTTPRequestField accepts the http.Request by value, so it's not mutated
+func HTTPRequestField(httpRequest http.Request) func(c *actions.FieldConfig) interface{} {
+	return func(c *actions.FieldConfig) interface{} {
+		header := make(map[string][]string, 0)
+		for k, v := range httpRequest.Header {
+			header[k] = v
+		}
+		body := ""
+		b, err := io.ReadAll(httpRequest.Body)
+		if err == nil {
+			body = string(b)
+		}
+
+		form := make(map[string][]string)
+		postForm := make(map[string][]string)
+		err = httpRequest.ParseForm()
+		if err == nil {
+			for k, v := range httpRequest.Form {
+				form[k] = v
+			}
+			for k, v := range httpRequest.PostForm {
+				postForm[k] = v
+			}
+		}
```

Comment:
can be omitted, just use the parsed forms in the struct below
type conversion will happen automatically